bridge/doublepuppet.go

```diff
@@ -118,30 +124,34 @@ var (
 const useConfigASToken = "appservice-config"
 const asTokenModePrefix = "as_token:"
 
+func (dp *doublePuppetUtil) CanAutoDoublePuppet(mxid id.UserID) bool {
+	_, hasSecret := dp.getLoginSecret(mxid)
+	return hasSecret
+}
+
 func (dp *doublePuppetUtil) Setup(ctx context.Context, mxid id.UserID, savedAccessToken string, reloginOnFail bool) (intent *appservice.IntentAPI, newAccessToken string, err error) {
 	if len(mxid) == 0 {
 		err = ErrNoMXID
 		return
 	}
-	_, homeserver, _ := mxid.Parse()
-	loginSecret, hasSecret := dp.br.Config.Bridge.GetDoublePuppetConfig().SharedSecretMap[homeserver]
-	// Special case appservice: prefix to not login and use it as an as_token directly.
-	if hasSecret && strings.HasPrefix(loginSecret, asTokenModePrefix) {
-		intent, err = dp.newIntent(ctx, mxid, strings.TrimPrefix(loginSecret, asTokenModePrefix))
-		if err != nil {
-			return
-		}
-		intent.SetAppServiceUserID = true
-		if savedAccessToken != useConfigASToken {
-			var resp *mautrix.RespWhoami
-			resp, err = intent.Whoami(ctx)
-			if err == nil && resp.UserID != mxid {
-				err = ErrMismatchingMXID
+	loginSecret, hasSecret := dp.getLoginSecret(mxid)
+	if savedAccessToken == "" || savedAccessToken == useConfigASToken {
```

Comment:
This check seems to have moved to the wrong place, the old code was fine

Reply:
The goal of this was to preserve a generated/custom token if there was one, instead of replacing it.  But the utility of that is negligible if non-as_token methods are deprecated.  I'll just revert this.